### PR TITLE
Upgrade to graphql-java 14.0

### PR DESCRIPTION
**Pull request by lburja**
This patch addresses issue #354 . It upgrades graphql-java to version 14, and fixes a compilation problem.

### CODE REVIEW

File: src/main/kotlin/com/coxautodev/graphql/tools/SchemaParser.kt [hunk 1]
```diff
@@ -128,7 +128,8 @@ class SchemaParser internal constructor(scanResult: ScannedSchemaObjects, privat
 
         val additionalObjects = objects.filter { o -> o != query && o != subscription && o != mutation }
 
-        return SchemaObjects(query, mutation, subscription, (additionalObjects + inputObjects + enums + interfaces + unions).toSet(), codeRegistryBuilder)
+        @Suppress("UNCHECKED_CAST") val dictionary = (additionalObjects + inputObjects + enums + interfaces + unions).toSet() as Set<GraphQLType>
```
Comment (vojtapol) [inline src/main/kotlin/com/coxautodev/graphql/tools/SchemaParser.kt]: We can get rid off the warning suppression in various ways. I would recommend something like this:

```suggestion
        val types = (objects as Set<GraphQLType>) + inputObjects + enums + interfaces + unions

        return SchemaObjects(query, mutation, subscription, types, codeRegistryBuilder)
```

Comment (lburja) [reply src/main/kotlin/com/coxautodev/graphql/tools/SchemaParser.kt]: The suggested code produces the same warning - "Unchecked cast":

Screenshot 2020-01-21 at 22 02 12

Comment (vojtapol) [reply src/main/kotlin/com/coxautodev/graphql/tools/SchemaParser.kt]: Right, it needs to be:
```
val types = (objects.toSet() as Set<GraphQLType>) + inputObjects + enums + interfaces + unions
```

Comment (lburja) [reply src/main/kotlin/com/coxautodev/graphql/tools/SchemaParser.kt]: Done. Thanks!

**Comment by lburja** [comment]
@oliemansm : Do you think this pull request can be merged? I'd like to keep up-to-date with the latest graphql-java library in my project, but for that, a release of graphql-java-tools is needed.

If this pull request still needs work, or there are other concerns, please let me know

**Pull request merged**